System/Process/Posix.hs

```diff
@@ -136,7 +136,9 @@ createProcess_Internal fun
    maybeWith withFilePath mb_cwd $ \pWorkDir ->
    maybeWith with mb_child_group $ \pChildGroup ->
    maybeWith with mb_child_user $ \pChildUser ->
-   withMany withFilePath (cmd:args) $ \cstrs ->
+   withFilePath cmd $ \cmdstr ->
+   withMany withCString args $ \argstrs -> do
```

Comment:
On Windows this would be wrong, it needs to be `withCWString` otherwise it's encoded as 8-byte string rather than 16-bytes.

Reply:
@Mistuke Thanks for the feedback; Then `withCWString` is indeed the correct choice as I suspected here: https://github.com/haskell/process/issues/319#issuecomment-2184005410

Reply:
What is the conclusion of this exchange? On Windows it would be wrong, but it's not Windows, right? It's POSIX. So is `withCString` indeed right in this case?

Reply:
> What is the conclusion of this exchange? On Windows it would be wrong, but it's not Windows, right? It's POSIX. So is `withCString` indeed right in this case?

Windows is already correct:

https://github.com/haskell/process/blob/302b43a2e0c514a6ad86cbc402c60bf7f86878e2/System/Process/Windows.hsc#L185

This PR can be merged.